packages/client/src/graphql/Foretold__GraphQL__GetAgentMeasurements.re

```diff
@@ -0,0 +1,183 @@
+                 startCursor
+                 endCursor
+               }
+               edges {
+                 node @bsRecord{
+                  id
+                  createdAt @bsDecoder(fn: "E.J.toMoment")
+                  relevantAt @bsDecoder(fn: "E.J.O.toMoment")
+                  description
+                  competitorType
+                  taggedMeasurementId
+                  value {
+                    floatCdf {
+                      xs
+                      ys
+                    }
+                    floatPoint
+                  }
+                  measurable: Measurable @bsRecord{
+                    id
+                    name
+                    expectedResolutionDate @bsDecoder(fn: "E.J.O.toMoment")
+                    state @bsDecoder(fn: "Context.Primary.MeasurableState.fromEnum")
+                    stateUpdatedAt @bsDecoder(fn: "E.J.O.toMoment")
+                  }
+
+                 }
+               }
+            }
+        }
+    }
+  |}
+];
+
+module QueryComponent = ReasonApollo.CreateQuery(Query);
+
+let toMesuarements = (measurements: array(node)) => {
+  measurements
+  |> Array.map(n =>
+       Context.Primary.Measurement.make(
+         ~id=n.id,
+         ~value=n.value |> MeasurementValue.decodeGraphql,
+         ~description=n.description,
+         ~createdAt=Some(n.createdAt),
+         ~competitorType=n.competitorType,
+         ~relevantAt=n.relevantAt,
+         ~measurableId=
+           switch (n.measurable) {
+           | Some(n) => Some(n.id)
+           | None => None
+           },
+         ~measurable=
+           switch (n.measurable) {
+           | Some(m) =>
+             Some(
+               Context.Primary.Measurable.make(~id=m.id, ~name=m.name, ()),
+             )
+           | None => None
+           },
+         (),
+       )
+     );
+};
+
+let unpackConnection = responseResult => {
+  let agent = responseResult##agent;
+
+  let measurementsEdges: option(array(node)) =
+    agent |> Rationale.Option.fmap(agent => agent.measurements |> unpackEdges);
+
+  let measurements =
+    measurementsEdges
+    |> Rationale.Option.fmap(toMesuarements)
+    |> E.A.O.defaultEmpty;
+
+  // @todo:
+  let pageInfo =
+    Context.Primary.PageInfo.fromJson({
+      "endCursor": Some("end"),
+      "hasNextPage": false,
+      "hasPreviousPage": false,
+      "startCursor": Some("start"),
+    });
+
+  Context.Primary.Connection.make(
+    ~pageInfo,
+    ~total=Some(10),
+    ~edges=measurements,
+  );
+};
+
+let componentMaker = (query, innerComponentFn) =>
+  QueryComponent.make(~variables=query##variables, o =>
+    o.result
+    |> E.HttpResponse.fromApollo
+    |> E.HttpResponse.fmap(unpackConnection)
+    |> innerComponentFn
+  )
+  |> E.React.el;
+
+type direction = Context.Primary.Connection.direction;
+
+type inputType('a) =
+  (
+    ~id: string,
+    ~first: int=?,
+    ~last: int=?,
+    ~after: string=?,
+    ~before: string=?,
+    unit
+  ) =>
+  'a;
```

Comment:
In future stories, we should probably focus more on drying some of this stuff up.

Reply:
Ok, got it.